File: Programming Fundamentals/Training/09. Sets And Dictionaries - Adv C# May 2016/Phonebook.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

class Phonebook
{
    static void Main(string[] args)
    {
        Dictionary<string, string> phonebook = FillPhonebook();
        SearchForContacts(phonebook);
    }

    private static void SearchForContacts(Dictionary<string, string> phonebook)
    {
        while (true)
        {
            var contact = Console.ReadLine();
            if (contact.ToLower() == "stop") break;

            if (phonebook.ContainsKey(contact))
            {
                Console.WriteLine($"{contact} -> {phonebook[contact]}");
            }
            else
            {
                Console.WriteLine($"Contact {contact} does not exist.");
            }
        }
    }

    private static Dictionary<string, string> FillPhonebook()
    {
        var phonebook = new Dictionary<string, string>();
        while (true)
        {
            string[] line = Console.ReadLine().Split('-');
            if (line.Contains("search")) break;

            string name = line[0];
            string number = line[1];

            if (phonebook.ContainsKey(name))
            {
                phonebook[name] = number;
            }
            else
            {
                phonebook.Add(name, number);
            }
        }

        return phonebook;
    }
}

File: Java_Fundamentals_January_2017/Java_Advanced/Exersices/04. Sets and Maps/src/Phonebook.java
import java.util.HashMap;
import java.util.Scanner;

public class Phonebook {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        HashMap<String, String> phonebook = new HashMap<>();

        while (true){
            String[] contact = scan.nextLine().split("-");
            if (contact[0].toLowerCase().equals("search")) break;

            String name = contact[0];
            String number = contact[1];

            phonebook.put(name, number);
        }

        while (true){
            String contact = scan.nextLine();
            if (contact.toLowerCase().equals("stop")) break;

            if (phonebook.containsKey(contact)){
                System.out.printf("%s -> %s\n", contact, phonebook.get(contact));
            }
            else {
                System.out.printf("Contact %s does not exist.\n", contact);
            }
        }


    }
}
